lib/codeowners/cli/suggestion_builder.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+module Codeowners
+  module Cli
+    # Build a list of suggestions case the pattern is not matching.
+    #
+    # Case the user have `fzf` installed, it works building suggestions from
+    # `fzf`. See more on #fzf_query.
+    #
+    # Without `fzf` it tries to suggest patterns using fuzzy match search picking
+    # all the files from the parent folder of the current pattern.
+    #
+    # See more on #fuzzy_match_query.
+    class SuggestionBuilder
```

Comment:
I think that we could extract suggestion providers from this class: separate one for fzf and fuzzy match. SuggestionProvider will be responsible for building query (now in `*_query` methods) and giving suggestions (`suggest_*` methods). `SuggestionBuilder` will be deciding which provider pick basing on what SuggestionProvider classes return from `.available?` method.  For `FuzzySuggestionProvider` this method will be always true, and for `FzfSuggestionProvider` - it'll be based on `installed_fzf?`.

Reply:
I agree. I also thought about it but I tried to make it simpler with a single class because of the scope of the classes. I'll isolate them.